### DISCUSSION TITLE
UnsupportedApiVersion Elsa Workflow Verison 2.14

### DISCUSSION THREAD

**Discussion by Rathna314** [elsa-workflows/elsa-core#5631] [General]
Hello all,

I am getting the following error

"error":{"code":"UnsupportedApiVersion","message":"The HTTP resource that matches the request URI '[http://localhost:#####](http://localhost/#####)' does not support the API version '1.0'.","innerError":null}}

We have upgraded all our projects to .Net 8.0 and I am running Elsa workflow API 2.14.1 version on Windows. The ELSA workflow API works fine when run as an individual project but when I it as a service with other projects it gives me the UnsupportedApiVersion error. Please help of what I need to update. I already tried some solutions for Startup.cs and controller  which were answered online.

**Comment by glime-ai** [comment]
To address the issue of receiving an "UnsupportedApiVersion" error when running Elsa Workflow API as a service with other projects, you should ensure that the API versioning in your application is correctly configured. Since you are using .NET 8.0, it's important to note that the way services are registered and middleware is configured might have changed from previous versions.

Here are some steps you can take to troubleshoot and resolve the issue:

1. **Check API Versioning Configuration**: Ensure that your application has API versioning configured correctly. You might need to specify the default API version or ensure that the version "1.0" is accepted.

2. **Review Route Configuration**: Verify that the routes for your API endpoints are set up correctly and that they include the expected version number if API versioning is used.

3. **Inspect Service Registration**: Make sure that all necessary services for Elsa Workflow are registered in the `Program.cs` file. If you have custom configurations or services, ensure they are compatible with the version of Elsa Workflow you are using.

4. **Compatibility with .NET 8.0**: Since you've upgraded to .NET 8.0, ensure that the version of Elsa Workflow API you are using is compatible with .NET 8.0. You may need to check the official documentation or GitHub issues for compatibility notes.

5. **Check for Breaking Changes**: Review the release notes for Elsa Workflow API and .NET 8.0 to identify any breaking changes that could affect your application.

6. **Consult GitHub Issues**: Look for similar issues reported by other users on the Elsa Workflow GitHub repository. There might be discussions or solutions related to API versioning problems.

7. **Logging and Debugging**: Add more detailed logging to your application to capture the full error stack trace. This can provide more insights into where the issue is occurring.

8. **Seek Support**: If you are unable to resolve the issue, consider reaching out for support on the Elsa Workflow GitHub repository or community forums.

Remember to back up your code and test any changes in a development environment before applying them to your production system.

If you provide more specific details about your configuration and the exact steps you've taken, I may be able to offer more targeted advice.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
